lib/pluginJsonapi.js

```diff
@@ -25,31 +25,48 @@ function pluginJsonapi( server, options, next ) {
 
   /**
   * From a type and an array of ids, create a href to find things on
-  * There may also be context to pass on to the new url
-  * @param {String} type
-  * @param {Array} ids
-  * @param {Array} context
+  * There may also be context and or filters to pass on to the new url
+  * @param {Object} hrefParameters
+  * @param {Object} context
   */
-  function _buildHref( type, ids, context ) {
-    ids = _.isArray( ids ) ? ids.join( ',' ) : ids;
+  function _buildHref( hrefParameters, context ) {
+    var ids = _.isArray( hrefParameters.ids ) ? hrefParameters.ids.join( ',' ) : hrefParameters.ids;
 
     var queryString = '';
-    // build the url parts of context
-    if ( !_.isUndefined( context ) && !_.isUndefined( context[type] ) ) {
-      // get all the query object into a querystring using qs
-      var query = {
-        context: {}
-      };
-      query.context[type] = context[type];
-      queryString = qs.stringify( query );
+    var query = {};
+
+    if ( !_.isUndefined( context ) && !_.isUndefined( context[hrefParameters.type] ) ) {
+      // build the url parts of context
+      query.context = {};
+      query.context[hrefParameters.type] = context[hrefParameters.type];
+    }
+
+    if ( !_.isUndefined( hrefParameters.filter ) ) {
+      // add the filters to the query
+      query.filter = _.reduce( hrefParameters.filter, function( result, filterValues, filter ) {
+        // filterValues is an array of strings
+        // now we need to join the array to have a string
+        // so we can have a query string like "filter[productIds]=ID1,ID2"
+        if ( _.isArray( filterValues ) ) {
+          result[filter] = filterValues.join( ',' );
+        }
+        return result;
+      }, {} );
+    }
+
+    // get all the query object into a querystring using qs
+    queryString = qs.stringify( query );
+
+    if ( _.isUndefined( ids ) || (_.isString( ids ) && ids === '' ) ) {
```

Comment:
Do you need the _.isString here? ids === '' will detect if it is a string and empty as well